### PR TITLE
Add Multiattack Improvements for Monsters.json (Part 1 of 3)

### PR DESCRIPTION
**Pull request by caldane**
This is the isolated code from PR #144 pertaining only to multiattack options only for **Aboleth** to **Frost Giant**

This will be split up into 3 PR's to keep the lines of code down.

### CODE REVIEW

**Comment by bagelbits** [comment]
You'll want to rebase all three PRs against the branch you're merging against. I just updated it.

File: src/5e-SRD-Monsters.json [hunk 1]
```diff
@@ -8676,7 +9394,63 @@
...
+                "count": 1,
+                "type": "melee"
+              },
+              {
+                "name": "Claws",
+                "count": 1,
+                "type": "melee"
+              }
+            ],
+            [
```
Comment (bagelbits) [inline src/5e-SRD-Monsters.json:9418]: For these two, I'm wondering if it makes sense to add a `note` that these are only if Fire Breath is available?

File: src/5e-SRD-Monsters.json [hunk 2]
```diff
@@ -8810,7 +9584,19 @@
     "actions": [
       {
         "name": "Multiattack",
-        "desc": "The chuul makes two pincer attacks. If the chuul is grappling a creature, the chuul can also use its tentacles once."
+        "desc": "The chuul makes two pincer attacks. If the chuul is grappling a creature, the chuul can also use its tentacles once.",
```
Comment (bagelbits) [inline src/5e-SRD-Monsters.json]: We could probably use a `note` here too? That way we can add the grapple modifier too? Maybe? I know this one is a little weird.

File: src/5e-SRD-Monsters.json [hunk 3]
```diff
@@ -12602,7 +13609,91 @@
     "actions": [
       {
         "name": "Multiattack",
-        "desc": "The erinyes makes three attacks"
+        "desc": "The erinyes makes three attacks",
+        "options": {
```
Comment (bagelbits) [inline src/5e-SRD-Monsters.json]: Nice combinatorics! :D

Comment (bagelbits) [reply src/5e-SRD-Monsters.json]: Actually, another way we could do this is:
```
{
  "choose": 3,
  "from": [
    [
      {
        "name": "Longsword",
        "count": 1,
        "type": "melee"
      }
    ],
    [
      {
        "name": "Longbow",
        "count": 1,
        "type": "ranged"
      }
    ],
    [
      {
        "name": "Variant: Rope of Entanglement",
        "count": 1,
        "type": "ability"
      }
    ]
  ]
}
```

Comment (bagelbits) [reply src/5e-SRD-Monsters.json]: That utilizes the choose format a bit better and makes it much cleaner.

Comment (caldane) [reply src/5e-SRD-Monsters.json]: OMG, that seems so obvious now, well that is a far better syntax I will improve that

Comment (jbhaywood) [reply src/5e-SRD-Monsters.json]: Is being able to choose more than one of the same option valid for multi-attacks across all monsters? The idea I was going with originally was that you always choose one of the available options until you've hit the "choose: x" limit.

Comment (bagelbits) [reply src/5e-SRD-Monsters.json]: Not for all but it is for some. There are a few where it says it has to be unique. I'm not sure the best way to handle that case though.

Comment (jbhaywood) [reply src/5e-SRD-Monsters.json]: A flag on the options object, something like "choose_unique:" true/false?

**Comment by jbhaywood** [comment]
Just a general comment on the introduction of a "note" field. Doing that means this data is back to requiring code that knows how to parse english sentences, which is exactly what I was trying to get away from. This is an awesome resource, and I was using it to program monster AI for a video game, so anywhere that I couldn't rely on a known schema made that more difficult.

What about instead of an arbitrary "note" field, there was a "prerequisite" object? Something like...
```
{ 
  "type": "available_ability"
  "name: "Horror Nimbus"
}
```
or
```
{ 
  "type": "successful_attack"
  "name: "Tentacles"
}
```

And the "type" would be a small set of known values.

**Comment by bagelbits** [comment]
Hmmm.... that could be interesting. @ogregoire Do you have any thoughts on this?

**Comment by caldane** [comment]
@jbhaywood @bagelbits 
We could leave the note field in for a more human readable version, but I do like the idea of a `prerequisite` or `condition` property.  The problem is we are going to have to come up with things that mean something to computers because these conditions are not spelled out in the api.

Also I would suggest that to be a new issue, so we can move forward with what we have.

**Comment by ogregoire** [comment]
It makes sense.

But ideally, there would be an `option` structure available like this:

    "options": {
      "choose": 1,
      "from" : [
        {
          "value": [
            {
              "name": "Bite",
              "count": 1,
              "type": "melee"
            },
            {
              "name": "Horns",
              "count": 1,
              "type": "melee"
            },
            {
              "name": "Claws",
              "count": 1,
              "type": "melee"
            }
          ]
        },
        {
          "value": [
            {
              "name": "Fire Breath",
              "count": 1,
              "type": "ability"
            },
            {
              "name": "Horns",
              "count": 1,
              "type": "melee"
            },
            {
              "name": "Claws",
              "count": 1,
              "type": "melee"
            }
          ],
          "prerequisite": {
            "type": "available_ability",
            "name": "FireBreath"
          }
        }
      ]
    }

But it starts to be very structured and the structures are numerous. I personnally like it and would tend to go that way, but I understand practicality and the repetition if that's what you prefer, and then go towards the structure.

**Comment by caldane** [comment]
@jbhaywood @bagelbits @ogregoire 
So do we think that has to happen before this PR is approved.  Or can we make that a new issue and work on it when this one is done.  I am afraid the scope of this is getting hard to track.

**Comment by bagelbits** [comment]
Honestly, I'm okay with keeping `note` for now as a human readable way to handle it and cut an issue to make a better programmatic way to represent the data. @jbhaywood @ogregoire ?

**Comment by ogregoire** [comment]
I don't mind keeping "note" for the time being.

**Comment by jbhaywood** [comment]
Sounds good to me.

**Review by bagelbits — APPROVED** [review]
LGTM

**Pull request merged**